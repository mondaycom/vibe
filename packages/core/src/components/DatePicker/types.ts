import moment from "moment";

export type Moment = moment.Moment;

export enum FocusInput {
  startDate = "startDate",
  endDate = "endDate"
}

export enum Direction {
  next = "next",
  prev = "prev"
}

export interface RangeDate {
  /**
   * The start date of the range.
   */
  startDate: Moment | null;
  /**
   * The end date of the range.
   */
  endDate: Moment | null;
}
