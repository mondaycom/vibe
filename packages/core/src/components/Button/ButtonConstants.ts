import { SIZES } from "../../constants/sizes";

/**
 * @deprecated
 */
export enum ButtonType {
  PRIMARY = "primary",
  SECONDARY = "secondary",
  TERTIARY = "tertiary"
}

/**
 * @deprecated
 */
export enum ButtonColor {
  PRIMARY = "primary",
  POSITIVE = "positive",
  NEGATIVE = "negative",
  INVERTED = "inverted",
  ON_PRIMARY_COLOR = "on-primary-color",
  ON_INVERTED_BACKGROUND = "on-inverted-background",
  BRAND = "brand",
  FIXED_LIGHT = "fixed-light",
  FIXED_DARK = "fixed-dark"
}

export type Size = (typeof SIZES)[keyof typeof SIZES];

export const BUTTON_KIND_CLASS_NAMES = {
  PRIMARY: ""
};

export const BUTTON_ICON_SIZE = 20;

/**
 * @deprecated
 */
export enum ButtonInputType {
  BUTTON = "button",
  SUBMIT = "submit",
  RESET = "reset"
}
