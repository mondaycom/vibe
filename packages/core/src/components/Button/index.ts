export { default as Button, type ButtonProps } from "./Button";

export * from "./Button.types";
