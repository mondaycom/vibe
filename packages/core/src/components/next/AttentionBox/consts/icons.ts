import { Info, Check, Alert, Warning, Favorite } from "@vibe/icons";
import { AttentionBoxType } from "../AttentionBox.types";
import { SubIcon } from "../../../../types";

export const ATTENTION_BOX_DEFAULT_ICONS: Record<AttentionBoxType, SubIcon> = {
  primary: Info,
  success: Check,
  danger: Alert,
  warning: Warning,
  dark: Favorite
};
