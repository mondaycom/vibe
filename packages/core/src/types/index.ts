export * from "./events";
export * from "./SubIcon";
export type { default as VibeComponent } from "./VibeComponent";
export type { default as VibeComponentProps } from "./VibeComponentProps";
export * from "./withStaticProps";
export * from "./ArrayLastElement";
export * from "./SplitString";
export * from "./ElementContent";
export * from "./Colors";
