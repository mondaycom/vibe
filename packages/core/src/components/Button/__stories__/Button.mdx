import Button from "../Button";
import { Meta } from "@storybook/blocks";
import {
  BUTTON_GROUP,
  BADGE,
  SPLIT_BUTTON
} from "../../../storybook/components/related-components/component-description-map";
import { TipIconButton } from "./Button.stories.helpers";
import * as ButtonStories from "./Button.stories";

<Meta of={ButtonStories} />

# Button

Buttons allow users to trigger an action or event with a single click.
For example, you can use a button for allowing the functionality of submitting a form, opening a dialog, canceling an action,
or performing a delete operation.

<Canvas of={ButtonStories.Overview} />

### Import

```js
import { Button } from "@vibe/core";
```

## Props

<PropsTable />

## Usage

<UsageGuidelines
  guidelines={[
    "Buttons may contain icon, on the left or the right side",
    "Use 8 px spacing between buttons",
    "Replace text with a loader if action is submitted, but still processing",
    "Button width is set by it’s content, avoid changing it width",
    "Use only one primary button, and any remaining calls to action should be represented as lower emphasis buttons"
  ]}
/>

<TipIconButton />

## Accessibility

<UsageGuidelines
  guidelines={[
    <>
      Using an <code>id</code> is recommended for all instances to ensure proper label association.
    </>,
    <>
      Always provide descriptive text content for buttons, or use <code>ariaLabel</code> prop when buttons contain only
      icons.
    </>,
    <>
      Use <code>ariaLabel</code> prop when you need to provide a custom accessible name that differs from the visible
      button text or for icon-only buttons.
    </>,
    <>
      Use <code>ariaLabeledBy</code> prop when an external element provides the accessible name for the button.
    </>,
    <>
      Use <code>ariaHasPopup</code> prop when the button triggers a popup menu or dialog.
    </>,
    <>
      Use <code>ariaExpanded</code> prop to indicate when a popup triggered by the button is open or closed.
    </>,
    <>
      Use <code>ariaControls</code> prop to link the button to the element it controls.
    </>,
    <>
      Use <code>aria-describedby</code> prop when additional descriptive text is needed for the button.
    </>,
    <>
      Use <code>aria-pressed</code> prop for toggle buttons to indicate their current pressed state.
    </>
  ]}
/>

## Variants

### Button kinds

There can be more than one button in a screen, but to create the hierarchy of actions we need to use button kinds.

<Canvas of={ButtonStories.ButtonKinds} />

### Sizes

<Canvas of={ButtonStories.Sizes} />

### Disabled

Set disable button for something that isn’t usable. Use a tooltip on hover in order to indicate the reason of the disabled action.

<Canvas of={ButtonStories.Disabled} />

### States

<Canvas of={ButtonStories.States} />

### Positive and Negative

<Canvas of={ButtonStories.PositiveAndNegative} />

### Icons

<Canvas of={ButtonStories.Icons} />

## Do’s and Don’ts

<ComponentRules
  rules={[
    {
      positive: {
        component: <Button>Get started</Button>,
        description: "Use 1 or 2 words, no longer than 4 words, with fewer than 20 characters including spaces."
      },
      negative: {
        component: <Button>Get started and enjoy discount!</Button>,
        description: "Don’t use punctuation marks such as periods or exclamation points."
      }
    },
    {
      positive: {
        component: <Button>Get started</Button>,
        description: "Use sentence-case capitalization."
      },
      negative: {
        component: [<Button>Get Started</Button>, <Button>GET STARTED</Button>],
        description: "Don’t use title case captalization or all caps."
      }
    },
    {
      positive: {
        component: [<Button kind="tertiary">Cancel</Button>, <Button>Get started</Button>],
        description: "Use primary button as the main action , put the tertiary as the second option."
      },
      negative: {
        component: [<Button>Get started</Button>, <Button kind="secondary">Cancel</Button>],
        description: "Use primary button next to secondary."
      }
    },
    {
      positive: {
        component: [<Button kind="tertiary">Cancel</Button>, <Button>Get started</Button>],
        description: "Use active verbs or phrases that clearly indicate action."
      },
      negative: {
        component: [<Button kind="tertiary">Yes</Button>, <Button>No</Button>],
        description: "Use vague and generic labels that make the user read the dialog before taking action."
      }
    }
  ]}
/>

## Use cases and examples

### Loading state

<Canvas of={ButtonStories.LoadingState} />

### Success state

<Canvas of={ButtonStories.SuccessState} />

### On color state

<Canvas of={ButtonStories.OnColorStates} />

### Adjacent buttons

<Canvas of={ButtonStories.AdjacentButtons} />

## Related components

<RelatedComponents componentsNames={[SPLIT_BUTTON, BUTTON_GROUP, BADGE]} />
