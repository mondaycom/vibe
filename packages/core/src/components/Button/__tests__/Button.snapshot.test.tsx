import { vi, describe, it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import { noop as NOOP } from "lodash-es";
import Button from "../Button";
import { WhatsNew } from "@vibe/icons";
import { ButtonSize } from "../Button.types";

vi.mock("../../Icon/CustomSvgIcon/CustomSvgIcon", () => ({
  default: () => <div data-testid="custom-icon-mock" />
}));

describe("Button renders correctly", () => {
  it("renders correctly with empty props", () => {
    const tree = renderer.create(<Button>Button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with className", () => {
    const tree = renderer.create(<Button className="test">Button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with id", () => {
    const tree = renderer.create(<Button id="test">Button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with callbacks", () => {
    const tree = renderer
      .create(
        <Button onMouseDown={NOOP} onClick={NOOP} onBlur={NOOP} onFocus={NOOP} className="test">
          Button
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  Object.values(["small", "medium", "large"]).forEach(size => {
    it(`renders Button size- ${size}`, () => {
      const tree = renderer.create(<Button size={size as ButtonSize}>Button</Button>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  Object.values(Button.colors).forEach(color => {
    it(`renders Button color- ${color}`, () => {
      const tree = renderer.create(<Button color={color}>Button</Button>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  Object.values(Button.kinds).forEach(kind => {
    it(`renders Button kind- ${kind}`, () => {
      const tree = renderer.create(<Button kind={kind}>Button</Button>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  it("renders correctly with leftIcon", () => {
    const tree = renderer.create(<Button leftIcon={WhatsNew}>Button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with rightIcon", () => {
    const tree = renderer.create(<Button rightIcon={WhatsNew}>Button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with props change ui", () => {
    const tree = renderer
      .create(
        <Button noSidePadding rightFlat leftFlat marginLeft marginRight active>
          Button
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("a11y", () => {
    it("renders correctly with a11y props (false)", () => {
      const tree = renderer
        .create(
          <Button ariaLabel="text" ariaControls="area" ariaExpanded={false} ariaLabeledBy="id" ariaHasPopup={false}>
            Button
          </Button>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("renders correctly with a11y props (true)", () => {
      const tree = renderer
        .create(
          <Button ariaLabel="text" ariaControls="area" ariaExpanded={true} ariaLabeledBy="id" ariaHasPopup={true}>
            Button
          </Button>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
