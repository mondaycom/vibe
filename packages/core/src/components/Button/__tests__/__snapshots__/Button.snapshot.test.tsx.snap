// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly a11y renders correctly with a11y props (false) 1`] = `
<button
  aria-busy={false}
  aria-controls="area"
  aria-disabled={false}
  aria-expanded={false}
  aria-haspopup={false}
  aria-label="text"
  aria-labelledby="id"
  className="button sizeMedium kindPrimary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly a11y renders correctly with a11y props (true) 1`] = `
<button
  aria-busy={false}
  aria-controls="area"
  aria-disabled={false}
  aria-expanded={true}
  aria-haspopup={true}
  aria-label="text"
  aria-labelledby="id"
  className="button sizeMedium kindPrimary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button color- brand 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorBrand"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button color- fixed-dark 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorFixedDark"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button color- fixed-light 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorFixedLight"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button color- inverted 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorInverted"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button color- negative 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorNegative"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button color- on-inverted-background 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorOnInvertedBackground"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button color- on-primary-color 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorOnPrimaryColor"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button color- positive 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorPositive"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button color- primary 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button kind- primary 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button kind- secondary 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindSecondary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button kind- tertiary 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindTertiary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button size- large 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeLarge kindPrimary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button size- medium 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders Button size- small 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeSmall kindPrimary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders correctly with callbacks 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="test button sizeMedium kindPrimary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders correctly with className 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="test button sizeMedium kindPrimary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders correctly with empty props 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders correctly with id 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorPrimary"
  data-testid="button_test"
  data-vibe="Button"
  id="test"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders correctly with leftIcon 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  <svg
    aria-hidden={true}
    className="icon leftIcon noFocusStyle"
    data-testid="icon"
    data-vibe="Icon"
    fill="currentColor"
    height="20"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      clipRule="evenodd"
      d="M10.778 2.657c.972-.928 2.576-.86 3.544-.027a2.482 2.482 0 0 1 .78 2.59h1.039c.966 0 1.75.783 1.75 1.75v1.677a1.75 1.75 0 0 1-1.75 1.75h-.124V16a1.75 1.75 0 0 1-1.75 1.75H5.7A1.75 1.75 0 0 1 3.951 16v-5.603h-.12a1.75 1.75 0 0 1-1.75-1.75V6.97c0-.967.783-1.75 1.75-1.75h.911a2.483 2.483 0 0 1 .78-2.586c.97-.833 2.573-.9 3.546.027.381.365.65.782.848 1.09l.005.008.009-.013c.197-.307.466-.725.848-1.089Zm2.566 1.11c-.449-.386-1.176-.363-1.53-.025-.25.238-.431.519-.646.852l-.038.059c-.102.158-.231.359-.396.549h2.325l.042.002c.29.016.488-.14.57-.435a.983.983 0 0 0-.327-1.002Zm-6.843.004c.45-.386 1.177-.363 1.531-.024.25.237.43.518.646.851l.037.06c.102.157.232.358.396.548H6.786l-.042.002c-.29.016-.488-.139-.57-.435a.983.983 0 0 1 .327-1.002Zm2.733 5.126H3.83a.25.25 0 0 1-.25-.25V6.97a.25.25 0 0 1 .25-.25h5.404v2.177Zm1.502 0V6.72h5.405a.25.25 0 0 1 .25.25v1.677a.25.25 0 0 1-.25.25h-5.405ZM5.451 16v-5.603h3.781v5.853h-3.53a.25.25 0 0 1-.25-.25Zm5.284.25v-5.853h3.782V16a.25.25 0 0 1-.25.25h-3.532Z"
      fillRule="evenodd"
    />
  </svg>
  Button
</button>
`;

exports[`Button renders correctly renders correctly with props change ui 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorPrimary colorPrimaryActive undefined marginRight marginLeft rightFlat leftFlat noSidePadding"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button renders correctly renders correctly with rightIcon 1`] = `
<button
  aria-busy={false}
  aria-disabled={false}
  className="button sizeMedium kindPrimary colorPrimary"
  data-testid="button"
  data-vibe="Button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  Button
  <svg
    aria-hidden={true}
    className="icon rightIcon noFocusStyle"
    data-testid="icon"
    data-vibe="Icon"
    fill="currentColor"
    height="20"
    viewBox="0 0 20 20"
    width="20"
  >
    <path
      clipRule="evenodd"
      d="M10.778 2.657c.972-.928 2.576-.86 3.544-.027a2.482 2.482 0 0 1 .78 2.59h1.039c.966 0 1.75.783 1.75 1.75v1.677a1.75 1.75 0 0 1-1.75 1.75h-.124V16a1.75 1.75 0 0 1-1.75 1.75H5.7A1.75 1.75 0 0 1 3.951 16v-5.603h-.12a1.75 1.75 0 0 1-1.75-1.75V6.97c0-.967.783-1.75 1.75-1.75h.911a2.483 2.483 0 0 1 .78-2.586c.97-.833 2.573-.9 3.546.027.381.365.65.782.848 1.09l.005.008.009-.013c.197-.307.466-.725.848-1.089Zm2.566 1.11c-.449-.386-1.176-.363-1.53-.025-.25.238-.431.519-.646.852l-.038.059c-.102.158-.231.359-.396.549h2.325l.042.002c.29.016.488-.14.57-.435a.983.983 0 0 0-.327-1.002Zm-6.843.004c.45-.386 1.177-.363 1.531-.024.25.237.43.518.646.851l.037.06c.102.157.232.358.396.548H6.786l-.042.002c-.29.016-.488-.139-.57-.435a.983.983 0 0 1 .327-1.002Zm2.733 5.126H3.83a.25.25 0 0 1-.25-.25V6.97a.25.25 0 0 1 .25-.25h5.404v2.177Zm1.502 0V6.72h5.405a.25.25 0 0 1 .25.25v1.677a.25.25 0 0 1-.25.25h-5.405ZM5.451 16v-5.603h3.781v5.853h-3.53a.25.25 0 0 1-.25-.25Zm5.284.25v-5.853h3.782V16a.25.25 0 0 1-.25.25h-3.532Z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;
